keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -73,279 +75,320 @@
+        | grpc DriverGrantBucketAccess({           |
+        |      "name": "ba-$uuid",                 |
+        |      "bucketID": "br-$uuid",             |
+        |      "parameters": {                     |
+        |         "key": "value"                   |
+        |      }                                   |
+        | })                                       |
+        |------------------------------------------|
+    ```
+
+    Each BucketAccess is meant to map to a unique service account in the OSP. Once the requested privileges have been granted, a secret by the name specified in `credentialsSecretName` in the BucketClaim is created. The secret will reside in the namespace of the BucketClaim. The secret will contain either access keys or service account tokens based on the chosen authentication type. The format of this secret can be found [here](#bucketinfo)
+
+    If this call returns successfully, the sidecar sets `AccessGranted` condition to `True` in the BucketAccess.
 
 NOTE:
- - The secret will not be created until the credentials are generated/service account mappings are complete.
- - Within a namespace, one BucketAccess and secret pair is generally sufficient, but cases which may want to control this more granularly can use multiple.
- - The secret will be created with a finalizer that prevents it from being deleted until the associated bucketAccess is deleted.
+- The secret will not be created until the credentials are generated/service account mappings are complete.
+- Within a namespace, one BucketAccess and secret pair is generally sufficient, but cases which may want to control this more granularly can use multiple.
+- The secret will be created with a finalizer that prevents it from being deleted until the associated bucketAccess is deleted.
 
-## Attaching Bucket Information to Pods
+### Attaching Bucket Information to Pods
 
 The following stakeholders are involved in the lifecycle of attaching bucket information into pods:
 
- - Users  - specify bucket in the pod definition
+- Users  - specify bucket in the pod definition
 
 Once a valid BucketAccess is available (`AccessGranted` is `True`), pods can use them.
 
-###### 1. User creates pod with a projected volume pointing to the secret in BucketAccess
+1.  **User creates pod with a projected volume pointing to the secret in BucketAccess**
 
 The secret mentioned in the `credentialsSecretName` field of the BucketAccess should be set as a projected volume in the PodSpec. If either the Bucket provisioning is incomplete, or the access generation is incomplete - the pod will not run. It will wait indefinitely for those two conditions to be true.
 
+<!-- TODO: this example shows a problem that might arise from our current plan changes ...
+           what if a user mounts multiple buckets to the same pod?
+           they can refer to the secret fields individually, of course, and define unique names for each bucket's fields
+           but is that too much work? -->
```

Comment:
In the `PodSpec` below, I am seeing what could be a complication for our plans to change the secret format.

If users have multiple sets of credentials, they will have to do manual work to make sure their config vars (mounted either as env vars, files, or whatever) don't overlap with each other. 

At the end of the day, this is still something that k8s users have to be somewhat mindful of, so maybe it doesn't matter too much. But it's something to consider.

@shanduur I'm curious your thoughts

Reply:
In my opinion it is not different than the situation that happens now. If there are 2 secrets to be mounted, those will still have to be either renamed or mounted in different directories. I am in favour of this design, as this can be easily done at the manifest level without need for init containers or changes to application.

Reply:
hello I'm trying to use a COSI s3 bucket within a tekton pipeline and this is what they have for awscli example https://github.com/tektoncd/catalog/tree/main/task/aws-cli/0.2 

I could totally work with this enhancement as well because awscli takes ENV_VARS I can set in my containers from the ones exposed by the new design, old design is troublesome as it's json in a secret property